### PR TITLE
[REBASE&FF]Add cbor coswid rim generation

### PR DESCRIPTION
**Pull request by Javagedes**
## Description

- Adds a rust crate to generate a [RIM](https://trustedcomputinggroup.org/wp-content/uploads/TCG-Reference-Integrity-Manifest-RIM-Information-Model-Version-1.1-Revision-1.0_pub-1.pdf) for the STM.

- Updates the target of MU_BASECORE for CI form feature/stm to release/202311 as the tool for generating the stm has been added to BaseTools in release/202311.

- Fixes a few uncrustify changes.

For details on how to complete to complete these options and their meaning refer to [CONTRIBUTING.md](https://github.com/microsoft/mu/blob/HEAD/CONTRIBUTING.md).

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Tested correct generation on platform.

## Integration Instructions

Platform must call `self.Helper.generate_stm_rim(...)` to generate the RIM.

### CODE REVIEW

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,464 @@
+//! A library containing necessary data structures representing CBOR formatted
```
Comment (makubacki) [inline SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs:1]: Can you please add a copyright?

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+            file: file.into(),
+            name: name.into(),
+            size,
+            alg: CoseAlgorithmPayload {
+                algorithm_id: algorithm,
+                digest: digest.into(),
+            },
+        }
+    }
+}
```
Comment (makubacki) [inline SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs:464]:
```suggestion
}

```

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        "F".repeat(256),
+    );
+
+    let mut buffer = Vec::new();
+    minicbor::encode(&cosesign1, &mut buffer).unwrap();
+    let mut file = std::fs::File::create(args.output.unwrap_or(PathBuf::from("RIM.bin"))).unwrap();
+    file.write_all(&buffer).unwrap();
+    Ok(())
+}
+
```
Comment (makubacki) [inline SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs:58]: Was there a particular reason for so many newlines here?

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+
+pub fn generate_measurements(file: &PathBuf) -> Result<Vec<FileMeasurement>> {
```
Comment (makubacki) [inline SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs:61]: Does this need to be public?

**Review by makubacki — APPROVED** [review]

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/validation.rs [hunk 5]
```diff
@@ -1,4 +1,11 @@
 //! A module containing the objects used for working with Validation rules.
+//! 
+//! ## License
+//!
+//! Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (kuqin12) [inline SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/validation.rs:5]:
```suggestion
//! Copyright (c) Microsoft Corporation.
```

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/util.rs [hunk 6]
```diff
@@ -1,4 +1,11 @@
 //! A module containing utility functions for extracting information from the PDB file.
+//! 
+//! ## License
+//!
+//! Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (kuqin12) [inline SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/util.rs:5]:
```suggestion
//! Copyright (c) Microsoft Corporation.
```

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/main.rs [hunk 7]
```diff
@@ -1,3 +1,9 @@
+//! ## License
+//!
+//! Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (kuqin12) [inline SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/main.rs:3]:
```suggestion
//! Copyright (c) Microsoft Corporation.
```

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs (hunk 1 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs:1]: Added to this, and multiple other files.

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs (hunk 2 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs]: Updated thanks

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/auxgen.rs [hunk 8]
```diff
@@ -1,5 +1,12 @@
 //! A module that contains the C-equivalent structs and final functions to
 //! generate the aux file.
+//! 
+//! ## License
+//!
+//! Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (kuqin12) [inline SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/auxgen.rs:6]:
```suggestion
//! Copyright (c) Microsoft Corporation.
```

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs (hunk 4 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs]: Nope. everything in main does not. I've removed.

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs [hunk 9]
```diff
@@ -0,0 +1,470 @@
+//! A library containing necessary data structures representing CBOR formatted
+//! COSE objects.
+//! 
+//! ## License
+//!
+//! Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (kuqin12) [inline SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs:6]: same here.

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs (hunk 3 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs:62]: nope. I did not run cargo fmt.

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs [hunk 10]
```diff
@@ -0,0 +1,109 @@
+//! ## License
+//!
+//! Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (kuqin12) [inline SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs:3]: And other instances in this PR.

**Review by kuqin12 — APPROVED** [review]

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs (hunk 3 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs:67]: updated now though.

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/util.rs (hunk 6 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/util.rs:5]: Gah. that will be all over the place. I'll fix.

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/validation.rs (hunk 5 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/validation.rs]: Updated

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/main.rs (hunk 7 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/main.rs]: Updated

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs (hunk 9 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/lib.rs]: Updated

File: SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs (hunk 10 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_rim/src/main.rs]: Updated

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/util.rs (hunk 6 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/util.rs]: Updated

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/main.rs [hunk 11]
```diff
@@ -1,3 +1,9 @@
+//! ## License
+//!
+//! Copyright () Microsoft Corporation.
```
Comment (makubacki) [inline SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/main.rs:3]:
```suggestion
//! Copyright (c) Microsoft Corporation.
```

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/validation.rs [hunk 12]
```diff
@@ -1,4 +1,11 @@
 //! A module containing the objects used for working with Validation rules.
+//! 
+//! ## License
+//!
+//! Copyright () Microsoft Corporation.
```
Comment (makubacki) [inline SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/validation.rs:5]:
```suggestion
//! Copyright (c) Microsoft Corporation.
```

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/main.rs (hunk 11 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/main.rs]: Fixed

File: SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/validation.rs (hunk 12 above)
Comment (Javagedes) [reply SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/validation.rs]: Fixed

**Pull request merged**